apps/aehttp/test/aehttp_integration_SUITE.erl

```diff
@@ -0,0 +1,657 @@
+                                   BlocksToCheck),
+    TopHeader = rpc(aec_conductor, top_header, []), 
+    BlocksToCheck = aec_headers:height(TopHeader),
+    lists:foreach(
+        fun(Height) ->
+            {ok, ExpectedBlock} = rpc(aec_conductor, get_block_by_height, [Height]),
+            {ok, H} = aec_blocks:hash_internal_representation(ExpectedBlock),
+            Hash = base64:encode(H),
+            ExpectedBlockMap = block_to_endpoint_map(ExpectedBlock), 
+            {ok, 200, BlockMap} = get_block_by_hash(Hash),
+            ct:log("ExpectedBlockMap ~p, BlockMap: ~p", [ExpectedBlockMap,
+                                                         BlockMap]),
+            % uncomment when fix aec_blocks:serialize_tx to have
+            % <<"tx">> instead of 'tx'
+            %BlockMap = ExpectedBlockMap,
+            #{<<"height">> := Height} = BlockMap
+        end,
+        lists:seq(0, BlocksToCheck)), % from genesis
+    ok.
+
+%% Maybe this test should be broken into a couple of smaller tests
+%% it currently tests the possitive cases for
+%% GET externalAPI/transactions
+%% POST internalAPI/spend-tx
+%% GET externalAPI/account/balance
+pending_transactions(_Config) ->
+    {ok, []} = rpc(aec_tx_pool, peek, [infinity]), % empty 
+    {ok, 200, []} = get_transactions(), 
+    {ok, 404, #{<<"reason">> := <<"Account not found">>}} = get_balance(),
+    BlocksToMine = 10,
+    AmountToSpent = 7,
+    Fee = 2,
+    MineReward = rpc(aec_governance, block_mine_reward, []),
+    aecore_suite_utils:mine_blocks(aecore_suite_utils:node_name(?NODE),
+                                   BlocksToMine),
+    {ok, 200, #{<<"balance">> := Bal0}} = get_balance(),  
+
+    Bal0 = BlocksToMine * MineReward,
+    true = (is_integer(Bal0) andalso Bal0 > AmountToSpent + Fee),
+
+    {ok, []} = rpc(aec_tx_pool, peek, [infinity]), % still empty 
+    {ok, 200, []} = get_transactions(), 
+
+    %{ok, SenderPubKey} = aecore_suite_utils:rpc_call(?NODE, aec_keys, pubkey, []),
+    ReceiverPubKey = random_hash(),
+    {ok, 404, #{<<"reason">> := <<"Account not found">>}} =
+                  get_balance(base64:encode(ReceiverPubKey)),
+
+    {ok, 200, _} = post_spend_tx(ReceiverPubKey, AmountToSpent, Fee),
+    {ok, NodeTxs} = rpc(aec_tx_pool, peek, [infinity]),
+    true = length(NodeTxs) =:= 1, % not empty anymore
+    {ok, 200, ReturnedTxs} = get_transactions(), 
+    ExpectedTxs = [#{<<"tx">> => base64:encode(aec_tx_sign:serialize_to_binary(T))}
+           || T <- NodeTxs],
+    true = length(ExpectedTxs) =:= length(ReturnedTxs),
+    true = lists:all(fun(Tx) -> lists:member(Tx, ExpectedTxs) end, ReturnedTxs),
+
+    {ok, 200, #{<<"balance">> := Bal0}} = get_balance(),  
+    {ok, 404, #{<<"reason">> := <<"Account not found">>}} =
+                  get_balance(base64:encode(ReceiverPubKey)),
+
+
+    % Currently aec_conductor:start fails to mine a new block;
+    % investigate this issue and uncomment the followin lines
```

Comment:
I will create a specific pivotal story to address this.

Reply:
In other tests we start conductro with automatic mining set to false.

Reply:
Mining is disabled by default here, too

Reply:
Will be fixed in [Pivotal story](https://www.pivotaltracker.com/story/show/153780170)